static/canvas.js

```diff
@@ -49,6 +51,7 @@ class Canvas {
         this.boxes = [];                // List of boxes in the current frame
         this.dragging = false;          // Keeps track of when we are dragging.
         this.enlargeDirection = null;   // Keeps the direction being dragged when enlarging.
+        this.previousSelection = null;  // Previous selection used for key frames.
```

Comment:
Is it better to have `previousSelection` and `selection` than `selection` and boolean `selected`?